lib/react_client.dart

```diff
@@ -587,7 +634,7 @@ final ReactDartInteropStatics2 _dartInteropStatics2 = (() {
 /// which produces a new JS [`ReactClass` component class](https://facebook.github.io/react/docs/top-level-api.html#react.createclass).
 @Deprecated('6.0.0')
 ReactDartComponentFactoryProxy _registerComponent(ComponentFactory componentFactory,
-    [Iterable<String> skipMethods = const []]) {
+    [Iterable<String> skipMethods = const ['getDerivedStateFromError', 'componentDidCatch']]) {
```

Comment:
These methods are not available in a V1 component are they? Also, without a call to `_filterSkipMethods`, I'm not sure what the point of this change is?

Reply:
I had originally included it because `_registerComponent` was allowing for `Component2` instances to go through it and was just forwarding those cases to `_registerComponent2`. So you're definitely right that for V1 it's not necessary, but if someone does use the method for `Component2` then having the defaults set here makes sure they don't accidentally set an error boundary (as an empty List default would have done). If I took `['getDerivedStateFromError', 'componentDidCatch']` away I could just not set a default value for `skipMethods` in `_registerComponent` at all so that `_registComponent2`'s defaults would stick. Would that be better?

Reply:
Ah, ok - forwarding the cases to `_registerComponent2` sounds good @joebingham-wk